app/browser/api/ledger.js

```diff
@@ -2347,21 +2348,8 @@ const initialize = (state, paymentsEnabled) => {
     }
   }
 
-  if (updateState.getUpdateProp(state, 'referralDownloadId') == null) {
-    promoCodeFirstRunStorage
-      .readFirstRunPromoCode()
-      .then((code) => {
-        onReferralCodeRead(code)
-      })
-      .catch(error => {
-        if (clientOptions.verboseP) {
-          console.error('read error: ' + error.toString())
-        }
-        fetchReferralHeaders()
-      })
-  } else {
-    fetchReferralHeaders()
-  }
+  // Schedule promo run and referral header fetch
+  schedulePromoRefFetch(state)
```

Comment:
I think we should move this call to the APP_INITIALIZED hook. in here:
https://github.com/brave/browser-laptop/blob/master/app/browser/reducers/ledgerReducer.js#L613
This gets called after most of the browser startup work and ledger.initialize and could save us a little bit on startup :) wdyt?

Reply:
I like it!

Reply:
@jasonrsadler done. Also I can confirm that I am receiving 400 for the promo PUT using the test code on master.